Fixed tag issues

shahidajmeri786 opened the pull request:
Fixed Shortcode tag.
Fixed Pre tag
Fixed img tag required figure tag
Fixed OL tag

shahidajmeri786 committed:
Skipped custom attributes for headers and p tags

shahidajmeri786 committed:
Fixed issue of span tag
Fixed issue of P tag
Fixed issue of headings tag

shahidajmeri786 committed:
Fixed tag issues

Fixed span tag issue
Fixed heading tag issue
Fixed p tag issue

shahidajmeri786 committed:
Fixed Shortcode tag.
Fixed Pre tag
Fixed img tag required figure tag
Fixed OL tag

shahidajmeri786 committed:
Merge branch 'development' of https://github.com/shahidajmeri786/convert-classic-editor-to-gb-blocks into development

# Conflicts:
#	class.convert-classic-to-gb-blocks.php

aslammultani commented on class.convert-classic-to-gb-blocks-admin.php:
```diff
@@ -103,7 +103,7 @@ public static function ccetgb_admin_bar_render() {
         global $post, $wp_admin_bar;
         if ( isset( $post ) && ! ( strpos( $post->post_content, "<!-- wp:" ) === false ) ){
-            return;
+          return;
```
There is mixed white space here. Can you correct this to use tabs instead of spaces?

aslammultani commented on class.convert-classic-to-gb-blocks.php:
```diff
@@ -88,6 +88,149 @@ public static function ccetgb_copy_as_new_draft()
+        $embed_provider = self::ccetgb_filter_iframeProvider($src);
+        $convert_content = self::ccetgb_convert_provider($embed_provider,$content);
+       return $convert_content;
+ }
```
Can you correct this to use tabs instead of spaces?

aslammultani commented on class.convert-classic-to-gb-blocks.php:
```diff
@@ -88,6 +88,149 @@ public static function ccetgb_copy_as_new_draft()
+    $provider_slug = '';
+  
+    $result = [];
+   if($src!==''){
```
use yoda condition, make code indentation throughout the function, add space in if condition between variable and operator

aslammultani commented on class.convert-classic-to-gb-blocks.php:
```diff
@@ -88,6 +88,149 @@ public static function ccetgb_copy_as_new_draft()
+     * @param string $content
+     * @param string $level
+     */
+    public static function ccetgb_convert_heading_tags($seperator='',$content='',$level=''){
```
Remove unused function

aslammultani commented on class.convert-classic-to-gb-blocks.php:
```diff
@@ -88,6 +88,149 @@ public static function ccetgb_copy_as_new_draft()
+        $src = str_replace("embed/","watch?v=",$src);
+        $provider_slug = 'youtube';
+       
+       
```
Use one line break instead of multi-line

aslammultani commented on class.convert-classic-to-gb-blocks.php:
```diff
@@ -276,7 +419,8 @@ public static function ccetgb_convert_classic_post_to_gb_blocks($content = '', $
                 }
                 if (strpos($content, '<div') !== false) {
-                    $content = str_replace('</div>', '</div><!-- /wp:md/multipurpose-gutenberg-block -->', $content);
+                 $content = preg_replace("/<div( [a-z][a-z0-9]*)[^>]*?(\/?)>/i",'<div class="wp-block-md-multipurpose-gutenberg-block is-block-center">', $content);
```
Make sure class and id attribute must be add.

aslammultani commented on class.convert-classic-to-gb-blocks.php:
```diff
@@ -300,43 +444,70 @@ public static function ccetgb_convert_classic_post_to_gb_blocks($content = '', $
                 if (strpos($content, '<p') !== false) {
-                    $content = str_replace('<p', '<!-- wp:paragraph --> <p', $content);
-                    $content = str_replace('</p>', '</p><!-- /wp:paragraph -->', $content);
+                    $content = preg_replace("/<p>/i",'<!-- wp:paragraph --> <p>', $content);
```
Make sure class attribute must be add dynamically throughout all element. Please reduce 3 line of code to two line code for replacing element. try following regex.
`/<p(|( [a-z][a-z0-9]*)[^>]*?(\/?))>/i`

aslammultani commented on class.convert-classic-to-gb-blocks.php:
```diff
@@ -300,43 +444,70 @@ public static function ccetgb_convert_classic_post_to_gb_blocks($content = '', $
-                    $content = str_replace('<h1', '<!-- wp:heading {"level":1} --><h1', $content);
-                    $content = str_replace("</h1>", '</h1><!-- /wp:heading -->', $content);
+                if (strpos($content, '<h1') !== false) {                    
+                    $content = preg_replace("/<h1>/i",'<!-- wp:heading {"level":1} --><h1>', $content);
```
Make sure class and id attribute must be add throughout h1 to h6.

aslammultani reviewed (changes requested):
Please make changes as I suggested below. And also make sure inside the `<div>` text must be wrap with <p> tag.

shahidajmeri786 committed:
<img> - Now working without figure tag.
<blockquote> - Working with visual editor support.
when used with combinations of tag in page  - Now working.

shahidajmeri786 committed:
Worked on <img> tag working without figure tag.
Worked on <blockquote> tag Working with visual editor support.
Worked on combination tags not working issue.
Worked on plain text should wrap inside p tag
Worked on <ol> tag support with ordered list
Worked on support class and id attributes to default gutenberg block attributes
Worked on header, section and shortcode not working issue
